content/chapters/io/lab/support/async/python/server.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+
+import sys
+import socket
+import logging
+
+
+def fibonacci(num):
+    if num in (0, 1):
+        return 1
+    return fibonacci(num-1) + fibonacci(num-2)
+
+
+def handle(connection, address):
+    logging.info("Received connection from %s", address)
+    response = ""
+    try:
+        msg = connection.recv(256).decode('utf-8')
+        logging.debug("Message: %s", msg)
+        num = int(msg)
+        if num < 0 or num > 34:
+            response = "out of range"
+        response = str(fibonacci(num))
+    except ValueError:
+        response = "error"
+
+    connection.send(response.encode('utf-8'))
+
+    connection.close()
+
+
+LISTEN_BACKLOG = 10
+
+def run_server(port, hostname="0.0.0.0"):
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    sock.bind((hostname, port))
+    sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
```

Comment:
```suggestion
    sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
    sock.bind((hostname, port))
```
`setsockopt()` must be called before `bind()`.